JavaScript/CodingConventions.md

```diff
@@ -477,11 +499,37 @@ function makeCookies(flavor) {
 
 We use JSDoc style comments above function declarations and at the top of files because they make code clear and easy to read, and we'd like to be able to generate JavaScript documentation at some point.
 
+The description for parameters is optional; description for `@returns` statement is not optional. Also it's encouraged to have description and params separated by newline.
```

Comment:
Consider this quoted text:
> When it's missing, I don't know if it's because the method doesn't return anything or if the
developer just forgot to include it. By requiring it to always be there,
you know for sure that if it's missing, it was a developer mistake. In
practice, I've found this to be extremely useful in avoiding
inconsistencies between the documentation comment and the code. Thus
`@returns {void}` as the shortest possible way to say that a function
doesn't return anything.

> Just to explain my position further: the goal is to ensure that the function doesn't get out of sync with the comment. It's trivial to accomplish that for named parameters, but harder to accomplish for return values because you have to poke at the AST to find return. So there are really just two options to accomplish my goal:

> 1. Everything has @returns all the time.
> 2. Only things that return values have @returns and everything else doesn't.
In considering 1, the solution is incredibly simple - just make sure the tag is there. Done.

> In considering 2, there are far more things to keep in mind:

> a. Should a function without return be required to not have @returns?
> b. What if a function sometimes returns a value and sometimes doesn't?
> c. Can you be certain when a function is inside of another function that you're keeping proper track?
> d. How much more traversing do you need to do to be sure of the presence of return or not?

> So implementing 1 is trivial to accomplish the goal while implementing 2 is, objectively, more work. Impossible? No. Just more work to accomplish the same thing.

Did it add enough clarification of purpose?